### PR TITLE
new protocol "homogenizer", and its test protocol ... 

### PR DESCRIPTION
**Pull request by mohsenkazemi**

### CODE REVIEW

File: pyworkflow/em/packages/xmipp3/protocol_align_volume.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import pyworkflow.protocol.params as params
 import pyworkflow.em as em
 from pyworkflow.em.packages.xmipp3.convert import getImageLocation
+from django.template.defaultfilters import default
```
Comment (delarosatrevin) [inline pyworkflow/em/packages/xmipp3/protocol_align_volume.py]: Why are you importing from Django here?

File: pyworkflow/em/packages/xmipp3/protocol_apply_transformation_matrix.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+        form.addParam('inputParticles', params.PointerParam, 
+                      pointerClass='SetOfParticles',
+                      pointerCondition='hasAlignmentProj',
+                      label="Input particles",  
+                      help="Aligned particles that their  "
+                           "angular assignment needs to be modified.")        
+        form.addParam('inputTransformMat', params.PathParam,
```
Comment (delarosatrevin) [inline pyworkflow/em/packages/xmipp3/protocol_apply_transformation_matrix.py]: This does not complain with our general design goals. We usually prefer to use file only at import level. Here probably are other ways how we can handle the input for this case.

File: pyworkflow/em/packages/xmipp3/protocol_align_volume.py [hunk 3]
```diff
@@ -27,6 +27,7 @@
 import pyworkflow.protocol.params as params
 import pyworkflow.em as em
```
Comment (mohsenkazemi) [inline pyworkflow/em/packages/xmipp3/protocol_align_volume.py]: test protocol updated

File: pyworkflow/em/packages/xmipp3/protocol_align_volume.py [hunk 4]
```diff
@@ -142,8 +143,10 @@ def _defineParams(self, form):
         line.addParam('initialShiftX', params.FloatParam, default=0, label='X')        
         line.addParam('initialShiftY', params.FloatParam, default=0, label='Y')
         line.addParam('initialShiftZ', params.FloatParam, default=0, label='Z')    
-             
-        group.addParam('initialScale', params.FloatParam, default=1, expertLevel=params.LEVEL_ADVANCED,
+        
+        group.addParam('optimizeScale', params.BooleanParam, default=False, expertLevel=params.LEVEL_ADVANCED,
```
Comment (mohsenkazemi) [inline pyworkflow/em/packages/xmipp3/protocol_align_volume.py]: necessary help added

File: pyworkflow/em/packages/xmipp3/protocol_apply_transformation_matrix.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
+# **************************************************************************
+# *
+# * Authors:     Mohsen Kazemi (mkazemi@cnb.csic.es)
+# *
```
Comment (mohsenkazemi) [inline pyworkflow/em/packages/xmipp3/protocol_apply_transformation_matrix.py]: related test protocol developed

File: software/em/xmipp/libraries/data/xmipp_image.h [hunk 6]
```diff
@@ -759,6 +759,33 @@ template<typename T>
             }
     }
 
+    /** flip image around Y axis
```
Comment (mohsenkazemi) [inline software/em/xmipp/libraries/data/xmipp_image.h]: These changes have been checked with @cossorzano some months ago!

File: software/em/xmipp/libraries/reconstruction/project.cpp [hunk 7]
```diff
@@ -117,6 +117,7 @@ void ProgProject::defineParams()
     addParamsLine("== Generating a set of projections == ");
     addParamsLine("  [--params <parameters_file>]           : File containing projection parameters");
     addParamsLine("                                         : Check the manual for a description of the parameters");
+    addParamsLine("                                         : tilt angle should be in the range [0-180]");
```
Comment (mohsenkazemi) [inline software/em/xmipp/libraries/reconstruction/project.cpp]: This has been checked with both @cossorzano and @rmarabini

File: software/em/xmipp/scipion_sconscript [hunk 8]
```diff
@@ -56,6 +57,14 @@ debug = get('DEBUG')
 matlab = get('MATLAB')
 opencv = env.GetOption('opencv') and get('OPENCV')
 
+if opencv:
+    opencvLibs = ['opencv_core',
```
Comment (mohsenkazemi) [inline software/em/xmipp/scipion_sconscript]: These modifications have been done with help of @rmarabini

**Pull request merged**